### PR TITLE
Prepare Public Docker Hub Image

### PR DESCRIPTION
**Pull request by bruno-cc**
### Description
This PR prepares the release of the public Docker Hub Image.
It adds a new documentation folder for keeping / reviewing the Docker Hub README + examples:
- `docker-compose.yml` -- a self-contained docker-compose deployment
- `example.sh` -- a script playing through Encryptonize's basic work flows using grpcurl & jq
- `README.md` -- README template for Docker Hub

The Docker Hub organization account is also prepared and shared between @philip-cc and me.
We can then tag an appropriate git commit and push a matching tag to Docker Hub.

#### Review:
- From encryption-service folder run `docker-compose build`, the docker image + tag should be ready
- You can proceed checking out the [docker-hub README](https://github.com/cyber-crypt-com/encryptonize-core/tree/public-docker-hub/documentation/docker-hub).
- Example Docker Hub page: https://hub.docker.com/repository/docker/brunocc/encryptonize-core

### Parent Issue
- https://github.com/cyber-crypt-com/encryptonize-core/issues/156

### Related Pull Requests
- NA

### Comments for the Reviewers
- Check if this README is the right level for Docker Hub
- Check if the commands / examples works

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [ ] Refactor
- [ ] Bugfix
- [x] Updated documentation
- [x] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [x] I have added/updated unit tests for functional changes.
- [x] I have added/updated integration tests for changes that affect dependent systems.
- [x] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [x] I have updated relevant workflows and deployment tools.
- [x] I have updated/added relevant documentation (readme, doc comments, etc).
- [x] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: documentation/docker-hub/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Encryptonize® Core is a secure cloud storage solution aimed at making it easy to manage sensitive
+data across multiple cloud environments while simultaneously removing the need to trust the cloud
+providers with encryption keys. Encryptonize® provides an Encryption Service with a simple gRPC
+interface that allows multiple workloads or users to store and retrieve data securely using cloud storage,
+leveraging S3 compatible storage like [Rook](https://rook.io/) with [Ceph](https://ceph.io/) and
+a PostgreSQL compatible database.
+
+## How to Use This Image
+The following Docker Compose file illustrates how to setup an example Encryptonize® deployment.
+It uses [MinIO](https://min.io/) as its objects store and [PostgreSQL](https://www.postgresql.org/) for auth storage.
```
Comment (philip-cc) [inline documentation/docker-hub/README.md:14]:
```suggestion
It uses [MinIO](https://min.io/) as its object store and [PostgreSQL](https://www.postgresql.org/) for auth storage.
```

File: documentation/docker-hub/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+**Warning**: The example just illustrates a quick way for trying out Encryptonize® and it not meant for any production usage!
```
Comment (philip-cc) [inline documentation/docker-hub/README.md:16]:
```suggestion
**Warning**: The example just illustrates a quick way for trying out Encryptonize® and is not meant for any production usage!
```

File: documentation/docker-hub/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      - 9000:9000
+    depends_on:
+      - auth-storage
+      - object-storage
+    tty: true
+
+  auth-storage:
+    image: postgres:13
+    environment:
+      POSTGRES_PASSWORD: password
```
Comment (philip-cc) [inline documentation/docker-hub/docker-compose.yml:44]: Might want to use the anchor here (like `MINIO_ACCESS_KEY` below)

File: encryption-service/docker-compose.yml [hunk 4]
```diff
@@ -46,6 +46,7 @@ services:
     build:
       context: .
       dockerfile: ./encryption-service.dockerfile
+    image: cybercryptcom/encryptonize-core:todo
```
Comment (philip-cc) [inline encryption-service/docker-compose.yml:49]: We should give this a proper tag and tag master as well.

File: documentation/docker-hub/docker-compose.yml (hunk 3 above)
Comment (bruno-cc) [reply documentation/docker-hub/docker-compose.yml]: ✔️

**Review by philip-cc — APPROVED** [review]
Ship it!

File: encryption-service/docker-compose.yml (hunk 4 above)
Comment (bruno-cc) [reply encryption-service/docker-compose.yml:49]: Will fix when publishing

**Pull request merged**